### PR TITLE
[#27] result parser for mets reader an others

### PR DESCRIPTION
**Pull request by pbroman**

### CODE REVIEW

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    public void setMapSeparator(String mapSeparator) {
+        this.mapSeparator = mapSeparator;
+    }
+
+    public void setValueConcatSeparator(String valueConcatSeparator) {
+        this.valueConcatSeparator = valueConcatSeparator;
+    }
+
+    /**
+     * Parses a list of strings and returns an map with the result.
```
Comment (ponchofiesta) [inline kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java]: typo: an map

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    @Override
+    public Map<String, String> parse(List<String> input) {
+
+        if (input == null) {
+            throw new IllegalArgumentException("The input list must not be null");
+        }
+
+        return input.stream()
+                .collect(Collectors.toMap(
+                    key -> StringUtils.trim(StringUtils.substringBefore(key, mapSeparator)),
```
Comment (ponchofiesta) [inline kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java]: Why not using `key -> StringUtils.substringBefore(key, mapSeparator).trim(),` ?

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.kitodo.mediaserver.core.api.IReadResultParser;
+
+
+/**
+ * A simple parser turning a list to a string-string-map.
+ */
+public class SimpleList2MapParser implements IReadResultParser {
+
+    private String mapSeparator = "=";
+    private String valueConcatSeparator = " ; ";
```
Comment (ponchofiesta) [inline kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java]: Why default separator with spaces?

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java (hunk 2 above)
Comment (pbroman) [reply kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java]: StringUtils.substringBefore: "A null string input will return null"

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java (hunk 3 above)
Comment (pbroman) [reply kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/SimpleList2MapParser.java]: Use case: several title fields that should be concatenated to one field

**Pull request merged**